src/extra/AutoUi/Collection/PersistentFilters.tsx

```diff
@@ -0,0 +1,184 @@
+	v: FilterSignature['value'];
+}
+
+export const isListQueryStringFilterRule = (
+	rule: any,
+): rule is ListQueryStringFilterObject =>
+	rule != null &&
+	typeof rule === 'object' &&
+	// it has to have an associated field
+	!!rule.n &&
+	typeof rule.n === 'string' &&
+	// it should at least have an operator
+	((!!rule.o && typeof rule.o === 'string') ||
+		// or a value
+		!!rule.v);
+
+export const listFilterQuery = (schema: JSONSchema, rules: JSONSchema[]) => {
+	const queryStringFilters = rules.map((filter) => {
+		// TODO: The typings need to be fixed in rendition.
+		const flatSchema = flattenSchema(schema);
+		const flattenFilter = flattenSchema(filter);
+		const signatures = decodeFilter(
+			flatSchema,
+			flattenFilter,
+		) as FilterSignature[];
+		return signatures.map<ListQueryStringFilterObject>(
+			({ field, operator, value }) => ({
+				n: field,
+				o: operator,
+				v: value,
+			}),
+		);
+	});
+	return qs.stringify(queryStringFilters);
+};
+
+const isQueryStringFilterRuleset = (
+	rule: any,
+): rule is ListQueryStringFilterObject[] =>
+	Array.isArray(rule) &&
+	!!rule?.length &&
+	rule?.every(isListQueryStringFilterRule);
+
+export const loadRulesFromUrl = (
+	searchLocation: string,
+	schema: JSONSchema,
+): JSONSchema[] => {
+	if (!searchLocation) {
+		return [];
+	}
+	const parsed = qs.parse(searchLocation.replace(/^\?/, '')) || {};
+	const rules = filter(parsed, isQueryStringFilterRuleset).map(
+		// @ts-expect-error
+		(rules: ListQueryStringFilterObject[]) => {
+			if (!Array.isArray(rules)) {
+				rules = [rules];
+			}
+
+			const signatures = rules.map(({ n, o, v }: any) => ({
+				field: n,
+				operator: o,
+				value: v,
+			}));
+			// full_text_search filter has always a single signature
+			// since we cannot add multiple search text filters.
+			// TODO: fix in rendition => this should be handled by Rendition, calling the
+			// createFilter function handle the case.
+			if (signatures[0].operator === FULL_TEXT_SLUG) {
+				return createFullTextSearchFilter(schema, signatures[0].value);
+			}
+			return createFilter(schema, signatures);
+		},
+	);
+
+	return rules;
+};
+
+interface PersistentFiltersProps extends FiltersProps {
+	viewsRestorationKey: string;
+	filtersRestorationKey: string;
+	history: History;
+}
+
+export const PersistentFilters = ({
+	schema,
+	views,
+	filters,
+	onViewsUpdate,
+	onFiltersUpdate,
+	viewsRestorationKey,
+	filtersRestorationKey,
+	history,
+	...otherProps
+}: PersistentFiltersProps) => {
+	const storedViews = React.useMemo(
+		() => getFromLocalStorage<FiltersView[]>(viewsRestorationKey) ?? [],
+		[viewsRestorationKey],
+	);
+	const urlRules = React.useMemo(
+		() => loadRulesFromUrl(history?.location?.search ?? '', schema),
+		[history?.location?.search, schema],
+	);
+	const storedFilters = React.useMemo(
+		() =>
+			!!urlRules?.length
+				? urlRules
+				: getFromLocalStorage<JSONSchema[]>(filtersRestorationKey) ?? [],
+		[urlRules, filtersRestorationKey],
+	);
```

Comment:
`urlRules` isn't atm used anywhere else, so how about v ?
```suggestion
	const storedFilters = React.useMemo(
		() => {
			const urlRules = loadRulesFromUrl(history?.location?.search ?? '', schema);
			return !!urlRules?.length
				? urlRules
				: getFromLocalStorage<JSONSchema[]>(filtersRestorationKey) ?? [];
		},
		[history?.location?.search, schema, filtersRestorationKey],
	);
```